### ISSUE TITLE
[Breaking Change Request] Remove support for ia32 from the dart sdk

### ISSUE THREAD

**Issue by a-siva** [dart-lang/sdk#59698]
### Intended Change
Remove support for ia32 from the dart-sdk completely starting with Dart SDK version 3.8.0. The ia32 distributions for all releases starting with 3.8.0 will be removed on dart.dev and other release channels.

### Rationale
This variant is not in use anymore

### Expected Impact
Users will not be able to install and run Dart SDK on ia32 architecture based systems.

### Projects known to break
None

### Projects that will not be affected
Flutter
All Google internal projects

### Migration Steps
Any project affected by this change needs to switch to the x64 artifacts or stay on Dart SDK version 3.7.0.

**Comment by a-siva** [comment]
//cc @mit-mit @athomas @itsjustkevin

**Comment by mit-mit** [comment]
Posted on Dart Announce

**Comment by mit-mit** [comment]
LGTM

**Comment by Piero512** [comment]
Will this remove support for Dart compiled to IA32? 

I had a minor concern over 32 bit DLLs which are common on Windows that will not be able to be connected through IPC by deploying a Dart solution.

Say, a Flutter app needs to interact with an old scanner library and can't load libtwain and the vendor library of the scanner on a different process written in dart, because Dart dropped IA-32 support, so it would need to be written in a different language that didn't drop x86 support.

Damn, there's so much legacy on Windows.

**Comment by a-siva** [comment]
> Will this remove support for Dart compiled to IA32?

Yes.

**Comment by a-siva** [comment]
> Say, a Flutter app needs to interact with an old scanner library and can't load libtwain and the vendor library of the scanner on a different process written in dart, because Dart dropped IA-32 support, so it would need to be written in a different language that didn't drop x86 support.

The different process written in Dart could still run using Dart 3.7

**Comment by itsjustkevin** [comment]
@mraleph @leonsenft and @johnmccutchan could you take a look at this breaking change request?

**Comment by mraleph** [comment]
LGTM

**Comment by johnmccutchan** [comment]
LGTM

**Comment by ntkme** [comment]
FYI @nex3

We need to change [cli_pkg](https://github.com/google/dart_cli_pkg/blob/main/lib/src/standalone/cli_platform.dart) slightly once this is rolled into 3.8.0-dev.

Also, I propose that we plan ahead and drop ia32 support for dart-sass at its 2.0 release, so that we don't have to worry about getting stuck at dart-sdk 3.7 in between dart-sass 2.0 and 3.0.

**Comment by nex3** [comment]
Yeah, that plan sounds reasonable to me. I'm not wholly opposed to just dropping support for ia32 ourselves without making it an official breaking change, just because our hands are tied by the upstream support, but it's definitely better to roll it into a major release since we have one coming anyway.

**Comment by mraleph** [comment]
I have received information about internal users which depend on X86 build, so I am marking this blocked for now - meaning that we can't proceed.

**Comment by mraleph** [comment]
After chatting with @mit-mit we have arrived to the conclusion that we can proceed with the breaking change and remove support from official builds of the dart sdk.

**Comment by mit-mit** [comment]
Breaking change approved

**Comment by mit-mit** [comment]
Updating dart.dev in https://github.com/dart-lang/site-www/pull/6390

**Comment by a-siva** [comment]
> After chatting with [@mit-mit](https://github.com/mit-mit) we have arrived to the conclusion that we can proceed with the breaking change and remove support from official builds of the dart sdk.

What about the internal test issue, has that been resolved ?

**Comment by mraleph** [comment]
> What about the internal test issue, has that been resolved ?

This is just about removing public Dart SDK builds. We will have to keep the code around until internal users have migrated away.

**Comment by a-siva** [comment]
> > What about the internal test issue, has that been resolved ?
> 
> This is just about removing public Dart SDK builds. We will have to keep the code around until internal users have migrated away.

Thanks for the clarification, this breaking request change was for complete removal of support for ia32 from the Dart SDK and it was marked as LGTM hence the confusion.

**Comment by parlough** [comment]
@mit-mit @mraleph Should the archive surface the builds for 3.7 as the final stable release with support or 3.6?

**Comment by mit-mit** [comment]
3.7.x as the last supported per https://github.com/dart-lang/sdk/issues/59698, please.
